### PR TITLE
[Data] Add writing benchmark

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

We don't have a release test that tracks the performance of writing data. This PR adds one.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by omatthew98 — APPROVED** [review]

File: release/nightly_tests/dataset/read_and_consume_benchmark.py [hunk 1]
```diff
@@ -58,6 +62,11 @@ def consume_fn(ds):
             for _ in ds.iter_internal_ref_bundles():
                 pass
 
+    elif args.write:
+
+        def consume_fn(ds):
+            ds.write_parquet(WRITE_PATH)
```
Comment (srinathk10) [inline release/nightly_tests/dataset/read_and_consume_benchmark.py:68]: We can probably append a UUID to write path and delete it, so we can run multiple copies of this script.

Comment (bveeramani) [reply release/nightly_tests/dataset/read_and_consume_benchmark.py:69]: Added a random prefix to the write path.

The S3 lifecycle policy is configured to delete objects everday.

File: release/nightly_tests/dataset/read_and_consume_benchmark.py [hunk 2]
```diff
@@ -58,6 +63,11 @@ def consume_fn(ds):
...
```
Comment (srinathk10) [inline release/nightly_tests/dataset/read_and_consume_benchmark.py:69]: Do we add a delete to cleanout these S3 objects written out? Don't want to accumulate a lot of data here.

Comment (bveeramani) [reply release/nightly_tests/dataset/read_and_consume_benchmark.py:69]: Yeah, I think they should automatically get cleaned out, but I'll monitor the bucket to confirm that the lifecycle policy works as expected

**Pull request merged**